File: benchmarkDS/ZeroShot/OPENAI_gpt-5-2025-08-07/Lse53.c
#include <stdio.h>

static const char* LOGIN_PATH = "/login";
static const char* USER_PATH  = "/user";
static const char* ADMIN_PATH = "/admin";

const char* secure_redirect(int is_logged_in, int is_admin) {
    if (!((is_logged_in == 0 || is_logged_in == 1) && (is_admin == 0 || is_admin == 1))) {
        return LOGIN_PATH;
    }
    if (is_logged_in == 0) {
        return LOGIN_PATH;
    }
    if (is_admin == 1) {
        return ADMIN_PATH;
    }
    return USER_PATH;
}

int main(void) {
    printf("%s\n", secure_redirect(0, 0)); // /login
    printf("%s\n", secure_redirect(1, 0)); // /user
    printf("%s\n", secure_redirect(1, 1)); // /admin
    printf("%s\n", secure_redirect(0, 1)); // /login
    printf("%s\n", secure_redirect(2, 1)); // /login (invalid input)
    return 0;
}

File: benchmarkDS/ZeroShot/OPENAI_gpt-5-2025-08-07/Lse53.java
import java.util.Objects;

public final class Lse53 {
    private Lse53() {}

    private static final String LOGIN_PATH = "/login";
    private static final String USER_PATH = "/user";
    private static final String ADMIN_PATH = "/admin";

    public static String secureRedirect(Boolean isLoggedIn, Boolean isAdmin) {
        if (isLoggedIn == null || isAdmin == null) {
            return LOGIN_PATH;
        }
        if (!Boolean.TRUE.equals(isLoggedIn)) {
            return LOGIN_PATH;
        }
        if (Boolean.TRUE.equals(isAdmin)) {
            return ADMIN_PATH;
        }
        return USER_PATH;
    }

    public static void main(String[] args) {
        System.out.println(secureRedirect(Boolean.FALSE, Boolean.FALSE)); // /login
        System.out.println(secureRedirect(Boolean.TRUE, Boolean.FALSE));  // /user
        System.out.println(secureRedirect(Boolean.TRUE, Boolean.TRUE));   // /admin
        System.out.println(secureRedirect(Boolean.FALSE, Boolean.TRUE));  // /login
        System.out.println(secureRedirect(null, Boolean.TRUE));           // /login
    }
}